src/reader/metadata.rs

```diff
@@ -22,29 +22,108 @@
 //! If they are compressed then their lengths indicate their
 //! compressed lengths.
 
+use std::collections::HashMap;
 use std::io::{Read, SeekFrom};
+use std::sync::Arc;
 
 use bytes::Bytes;
 use prost::Message;
 use snafu::{OptionExt, ResultExt};
 use tokio::io::{AsyncRead, AsyncReadExt, AsyncSeek, AsyncSeekExt};
 
 use crate::error::{self, Result};
-use crate::proto::{Footer, Metadata, PostScript, StripeFooter};
+use crate::proto::{self, Footer, Metadata, PostScript, StripeFooter};
 use crate::reader::decompress::Decompressor;
+use crate::statistics::ColumnStatistics;
+use crate::stripe::StripeMetadata;
 
 use super::decompress::Compression;
+use super::schema::{create_schema, TypeDescription};
 use super::ChunkReader;
 
 const DEFAULT_FOOTER_SIZE: u64 = 16 * 1024;
 
 /// The file's metadata.
 #[derive(Debug)]
 pub struct FileMetadata {
-    pub postscript: PostScript,
-    pub footer: Footer,
-    pub metadata: Metadata,
-    pub stripe_footers: Vec<StripeFooter>,
+    compression: Option<Compression>,
+    type_description: Arc<TypeDescription>,
+    number_of_rows: u64,
+    /// Statistics of columns across entire file
+    column_statistics: Vec<ColumnStatistics>,
+    stripes: Vec<StripeMetadata>,
+    user_custom_metadata: HashMap<String, Vec<u8>>,
+    // TODO: for now keeping this, but ideally won't want all stripe footers here
+    //       since don't want to require parsing all stripe footers in file unless actually required
+    stripe_footers: Vec<StripeFooter>,
```

Comment:
Hoping to move this out of this struct, as don't want to require parsing all stripe footers in file upfront (at least not in this metadata section)